pinecone/index_connection_test.go

```diff
@@ -22,93 +28,119 @@ import (
 // Integration tests:
 type IndexConnectionTestsIntegration struct {
 	suite.Suite
-	host             string
-	dimension        int32
-	apiKey           string
-	indexType        string
-	idxConn          *IndexConnection
-	sourceTag        string
-	idxConnSourceTag *IndexConnection
-	vectorIds        []string
+	host              string
+	dimension         int32
+	apiKey            string
+	indexType         string
+	idxConn           *IndexConnection
+	sourceTag         string
+	idxConnSourceTag  *IndexConnection
+	vectorIds         []string
+	client            *Client
+	podIdxName        string
+	serverlessIdxName string
 }
 
 func TestIntegrationIndexConnection(t *testing.T) {
 	apiKey := os.Getenv("PINECONE_API_KEY")
 	assert.NotEmptyf(t, apiKey, "PINECONE_API_KEY env variable not set")
 
-	client, err := NewClient(NewClientParams{ApiKey: apiKey})
-	if err != nil {
-		t.FailNow()
-	}
+	client, err := NewClient(NewClientParams{ApiKey: apiKey, Headers: map[string]string{"content-type": "application/json"}})
+	require.NotNil(t, client, "Client should not be nil after creation")
+	require.NoError(t, err)
 
 	podIndexName := os.Getenv("TEST_PODS_INDEX_NAME")
```

Comment:
Rather than hard-coding this name into an invisible secret in GitHub, I'd prefer if we could generate a random name to use each run since the suite is managing it's resources anyways.

If we hard-code things like this we'd also run into problems if we want to ever run integration tests in parallel, because the tests are no longer isolated and are all pointing at this predefined set of index names. 

Can we look at adding a helper function that generates a random name, possibly with a seed string? Something like we do in our other test suites:
- TypeScript: https://github.com/pinecone-io/pinecone-ts-client/blob/b5a66e6216d7615576dd881a67db627be2aa7bba/src/integration/test-helpers.ts#L92
- Python: https://github.com/pinecone-io/pinecone-python-client/blob/d9df37558ba35b7097dc4bc5c8101b576019caa2/tests/integration/helpers/helpers.py#L13

If we allow the test runs to create their own names and resources and then clean those up, that feels a bit more robust.

Reply:
Yes, fantastic idea!